include/mouse.h

```diff
@@ -65,7 +65,7 @@ enum class MouseMapStatus : uint8_t {
 // ***************************************************************************
 
 void MOUSE_EventMoved(const float x_rel, const float y_rel,
-                      const uint16_t x_abs, const uint16_t y_abs);
+                      const int32_t x_abs, const int32_t y_abs);
```

Comment:
Do these strictly operate on 32-bit integers?

If no, and when it no longer makes sense for a function to operate on exact type-size and signed values, you can relax the API back to just plain `integer`, which is a hint to readers that the function can handle general integers.

Reply:
I don't like the `int` type, as it is never obvious what range of values we expect. I used `int32_t`, because this is what SDL provides: https://wiki.libsdl.org/SDL_MouseMotionEvent . I would prefer not to increase the variable size, as we currently have no use for extra bits if `int` happens to be 64-bit.

Reply:
Makes sense; if it's to relay SDL `int32_t` values, then makes sense to hold that type-size (and not mix `int` with `int32_t`).

Reply:
Compilers on some very small controller chips will let `int` get down to 16-bit, like the Atmel Atmega. 

If you know the function should handle values to a specific size, then it also makes sense to set the type width. Certainly without our lifetime there will be displays having X and Y coordinates that exceed 16-bit.

Reply:
On the other end of the spectrum, if we know values should never programmatically (or API-wise) exceed a specific size, then it also makes sense to cap the type size.   

This is definitely why we've De-Bitu'd a lot of stuff was never 64-bit on DOS PCs. For example, in the upstream DOSBox, DOS IO ports are all 64-bit Bitus (in 64-bit architectures), but DOS only used 16-bit IO ports - so that entire subsystems has been De-Bitu'd.